liquigraph-core/src/main/java/org/liquigraph/core/configuration/ConfigurationBuilder.java

```diff
@@ -98,6 +102,15 @@ public ConfigurationBuilder withUsername(String username) {
     }
 
     /**
+     * Specifies the default username as the username allowed to connect to the remote
+     * graph database instance.
+     * @return itself for chaining purposes
+     */
+    public ConfigurationBuilder withDefaultUsername() {
```

Comment:
With the first [official beta release](https://neo4j.com/blog/neo4j-3-1-beta-release/) of Neo4j 3.1, we can see that Neo4j is going multi-user. There's still a single `neo4j` user by default at this point, but for how long?

My point is: isn't the explicit validation enough?

Besides, we'd be requiring that the user calls
```java
.withDefaultUsername()
```
instead of
```java
.withUsername("neo4j")
```
but these 2 calls have exactly the same length. I'd just keep the API simpler.

Convention over configuration is great, if you have sane defaults. But IMHO, you can't separate username and password: you either have defaults credentials (i.e. the pair) or you provide them completely. Of course, in our case, the default credentials are no credentials at all.